test/terra/backends/test_pulse_simulator.py

```diff
@@ -0,0 +1,302 @@
+
+        #Get a pulse configuration from the fake backend
+        self.backend_fake = FakeOpenPulse2Q()
+        self.back_config = self.backend_fake.configuration()
+        self.back_config_dict = self.back_config.to_dict()
+        self.system = pulse.PulseChannelSpec.from_backend(self.backend_fake)
+        self.defaults = self.backend_fake.defaults()
+
+        self.qubit = 0 # use 0 qubit for tests
+        self.freq_qubit = self.defaults.qubit_freq_est[self.qubit]
+
+        #Get pulse simulator backend
+        self.backend_sim = qiskit.Aer.get_backend('pulse_simulator')
+
+        self.drive_samples = 100
+
+    def create_qobj(self, shots, omega0, omega1, omega, phi, meas_level):
+        """ Creates qobj for the specified pulse experiment
+
+        Args:
+            See Hamiltonian at top of file
+            shots (int): number of times to perform experiment
+            omega0 (float): qubit frequency
+            omega1 (float): drive power/amplitude
+            omega (float): drive frequency
+            phi (float): drive phase
+            meas_level (int): how to return the data
+
+        Returns:
+            Qobj: qobj representing this pulse experiment
+        """
+
+        # Acquire pulse
+        acq_cmd = pulse.Acquire(duration=self.drive_samples)
+        acquire = acq_cmd(self.system.acquires, self.system.memoryslots)
+
+        # Create schedule
+
+        # drive pulse (just phase; omega1 included in Hamiltonian)
+        const_pulse = np.ones(self.drive_samples)
+        phase = np.exp(1j*phi)
+        drive_pulse = SamplePulse(phase*const_pulse, name='drive_pulse')
+
+        # add commands to schedule
+        schedule = pulse.Schedule(name='drive_pulse')
+        schedule += drive_pulse(self.system.qubits[self.qubit].drive)
+        schedule += acquire << schedule.duration
+
+        # Create the hamiltonian
+        hamiltonian = {}
+        hamiltonian['h_str'] = []
+
+        #Q0 terms
+        hamiltonian['h_str'].append('-0.5*omega0*Z0')
+        hamiltonian['h_str'].append('0.5*omega1*X0||D0')
+
+        hamiltonian['vars'] = {'omega0': omega0, 'omega1': omega1}
+
+        # set the qubit dimension to 2
+        hamiltonian['qub'] = {'0': 2}
+
+        # update the back_end
+        self.back_config_dict['hamiltonian'] = hamiltonian
+        self.back_config_dict['noise'] = {}
+        self.back_config_dict['dt'] = 1.0
+
+        self.back_config_dict['ode_options'] = {} # optionally set ode settings
+        self.back_config_dict['qubit_list'] = [0] # restrict qubit set
+
+        qobj = assemble([schedule], self.backend_fake,
+                        meas_level=meas_level, meas_return='single',
+                        memory_slots=1, qubit_lo_freq=[omega/(2*np.pi)],
+                        shots=shots, sim_config=self.back_config_dict)
+
+        return qobj
+
+    # ---------------------------------------------------------------------
+    # Test gates (using meas level 2)
```

Comment:
An option would be to just subclass the `TestPulseSimulator` into `TestPulseSimulatorMeasLevelOne` and `TestPulseSimulatorMeasLevelTwo`